examples/at_circular_piston_3D/at_circular_piston_3D.py

```diff
@@ -0,0 +1,244 @@
+sensor.record_start_index = kgrid.Nt - (record_periods * ppp) + 1
+
+# --------------------
+# SIMULATION
+# --------------------
+
+simulation_options = SimulationOptions(
+    pml_auto=True,
+    data_recast=True,
+    save_to_disk=True,
+    save_to_disk_exit=False,
+    pml_inside=False)
+
+execution_options = SimulationExecutionOptions(
+    is_gpu_simulation=True,
+    delete_data=False,
+    verbose_level=2)
+
+sensor_data = kspaceFirstOrder3D(
+    medium=deepcopy(medium),
+    kgrid=deepcopy(kgrid),
+    source=deepcopy(source),
+    sensor=deepcopy(sensor),
+    simulation_options=simulation_options,
+    execution_options=execution_options)
+
+# extract amplitude from the sensor data
+amp, _, _  = extract_amp_phase(sensor_data['p'].T, 1.0 / kgrid.dt, source_f0, dim=1, fft_padding=1, window='Rectangular')
+
+# reshape data
+amp = np.reshape(amp, (Nx - 1, Ny), order='F')
+
+# extract pressure on axis
+amp_on_axis = amp[:, Ny // 2]
+
+# define axis vectors for plotting
+x_vec = kgrid.x_vec[1:, :] - kgrid.x_vec[0]
+y_vec = kgrid.y_vec
+
+# =========================================================================
+# ANALYTICAL SOLUTION
+# =========================================================================
+
+# calculate the wavenumber
+k: float = 2.0 * np.pi * source_f0 / c0
+
+# define radius and axis
+a: float = source_diam / 2.0
+x_max: float = (Nx - 1) * dx
+delta_x: float = x_max / 10000.0
+x_ref: float = np.arange(0.0, x_max + delta_x, delta_x, dtype=float)
+
+# calculate the analytical solution for a piston in an infinite baffle
+# for comparison (Eq 5-7.3 in Pierce)
+r_ref = np.sqrt(x_ref**2 + a**2)
+p_ref = source_amp * np.abs(2.0 * np.sin((k * r_ref - k * x_ref) / 2.0))
+
+# get analytical solution at exactly the same points as k-Wave
+r = np.sqrt(x_vec**2 + a**2)
+p_ref_kw = source_amp * np.abs(2.0 * np.sin((k * r - k * x_vec) / 2.0))
+
+# calculate error
+L2_error = L2_error(p_ref_kw, amp_on_axis, ord=2)
+Linf_error = L2_error(p_ref_kw, amp_on_axis, ord=np.inf)
+# =========================================================================
+# VISUALISATION
+# =========================================================================
+
+# plot the pressure along the focal axis of the piston
+fig1, ax1 = plt.subplots(1, 1)
+ax1.plot(1e3 * x_ref, 1e-6 * p_ref, 'k-', label='Exact')
+ax1.plot(1e3 * x_vec, 1e-6 * amp_on_axis, 'b.', label='k-Wave')
+ax1.legend()
+ax1.set(xlabel='Axial Position [mm]',
+        ylabel='Pressure [MPa]',
+        title='Axial Pressure')
+ax1.set_xlim(0.0, 1e3 * axial_size)
+ax1.set_ylim(0.0, 1.1 * source_amp * 2e-6)
+ax1.grid()
+
+# plot the source mask (pml is outside the grid in this example)
+fig2, ax2 = plt.subplots(1, 1)
+ax2.pcolormesh(1e3 * np.squeeze(kgrid.x_vec),
+               1e3 * np.squeeze(kgrid.y_vec),
+               source.p_mask[:, :, int(np.ceil(Nz / 2))].T,
+               shading='gouraud')
+ax2.set(xlabel='y [mm]',
+        ylabel='x [mm]',
+        title='Source Mask')
+

```

Comment:
These plots seem to be rotated relative to the k-wave example and the axis is wrong

Reply:
same in the notebook